Automattic/woocommerce-subscriptions-core
# Upgrade to latest Node LTS v18

Fixes #230

## Description







## How to test this PR





1. Check out this branch.
2. Install the latest LTS version of node by running `nvm install` from this repo's directory.
3. Run `nvm use`. You should now be using Node v18.x.
4. Run `npm install`. You should not see an `Unsupported engine` warning.
5. Run `npm run build`. The build should complete successfully.

- [x] No `npm install` warning related to unsupported engines.
   - Note: 2 unsupported engine warnings occur: `@es-joy/jsdoccomment@0.20.1` `eslint-plugin-jsdoc@37.9.7`.
      These occur due to `@woocommerce/eslint-plugin` and `@wordpress/scripts`.
- [x] `npm run build` works as expected without errors.
   - Build artefacts in `./build` have minor changes, presumably due to updating `@wordpress/scripts`.
- [ ] `package-lock.json` reflects any changes related to the node/npm version upgrade.

## Product impact
